nautobot/dcim/signals.py

```diff
@@ -66,53 +67,52 @@ def handle_rackgroup_location_change(instance, created, **kwargs):
 
     Note that this is non-trivial for Location changes, since a LocationType that can contain RackGroups
     may or may not be permitted to contain Racks or PowerPanels. If it's not permitted, rather than trying to search
-    through child locations to find the "right" one, the best we can do is simply to null out the location.
+    through child locations to find the "right" one, the best we can do is to raise to raise a ValidationError
+    and roll back the changes we made.
     """
-    if not created:
-        if instance.location is not None:
+    with transaction.atomic():
+        if not created:
```

Comment:
Would it make sense to instead do `if created: return` first to reduce the amount of nesting of code here and in the next function?